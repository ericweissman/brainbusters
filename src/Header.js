import React, { Component } from 'react';
import Instructions from './Instructions'
import './styles/Main.scss'

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showInstructions: false
    }
  }

  toggleInstructionsShown = () => {
    this.setState({
      showInstructions: !this.state.showInstructions
    })
  }

  render() {
    if (this.state.showInstructions === false) {
      return (
        <header className="header">
          <h1><i className="fas fa-brain" />Brainbusters</h1>
          <div>
            <button className="toggle-instructions-btn"onClick={this.toggleInstructionsShown}>Show Instructions</button>
            <div>
              <select>
                <option value="show-all">Show All Questions</option>
                <option value="show-studylist">Study List Only</option>
              </select>
            </div>
          </div>
        </header>
      )
    } else {
      return (
        <header className="header">
          <h1><i className="fas fa-brain" />Brainbusters</h1>
          <div>
            <Instructions />
            <button className="toggle-instructions-btn" onClick={this.toggleInstructionsShown}>Got it!</button>
          </div>
        </header>
      )
    }
    
  }
}

export default Header;