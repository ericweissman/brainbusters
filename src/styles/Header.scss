header {
  color: #12c2e9;
  background: $dark;
  display: flex;
  justify-content: space-between;
  align-items: center;
  text-transform: uppercase;
  min-height: 125px;
 

  h1 {
    animation: color-change 12s infinite;
    padding-left: 50px;
    font-family: $title;
    font-size: 50px;
  }

  select {
    width: 248px;
    height: 30px;
    font-family: $text;
    font-size: 15px;
    outline: none;
    text-align: center;
    margin: 15px 0px 20px 0px;
  }

  button {
    @include instructions-button
  }
}

.toggle-instructions-btn {
  @include instructions-button;
}

.user-controls {
  display: flex;
  flex-direction: column;
  padding-right: 60px;
  align-items: center;
}

// ICON STYLE
.fas {
  color: $grad-title
}

@keyframes color-change {
  0% {
    color: #9F3FB2;
    transform: scale(1);
  }


  50% {
    color: #CB53A3;
    transform: scale(1.08);
  }

  100% {
    color: #9F3FB2;
    transform: scale(1);
  }
}

@media screen and (max-width: 1199px) {
  .user-controls {
    button {
      margin: 5px;
    }
  }
}

@media screen and (max-width: 1000px) {
  header {
    flex-direction: column;

    h1 {
      padding: 0;
      margin: 10px;
    }
  }
  .user-controls {
    padding: 0;
      button {
          margin: 5px 0;
          font-size: 15px;
          width: 155px;
      }
    padding-bottom: 20px;
  }

}

@media screen and (max-width: 500px) {
    header {
      h1 {
      font-size: 38px;
    }
  }
    .user-controls {
      padding-bottom: 10px;
        button {
            font-size: 12px;
            width: 135px;
       }
  }
}
  

@media screen and (max-width: 375px) {
    header {
      h1 {
      font-size: 33px;
      margin: 5px;
    }
  }
}