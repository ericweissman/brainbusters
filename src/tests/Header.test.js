import React from 'react';
import Header from '../Header';
import setupTests from '../setupTests'
import { shallow } from 'enzyme';


describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Header/>
    )
  })
  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the proper default state', () => {
    expect(wrapper.state()).toEqual({ showInstructions: false });
  });

  it('should change state when the Instructions button is clicks', () => {
    wrapper.find('.toggle-instructions-btn').simulate('click', { showInstructions: false })
    expect(wrapper.state('showInstructions')).toEqual(true)
  })
})

