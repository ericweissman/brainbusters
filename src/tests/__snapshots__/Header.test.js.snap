// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot when all data is passed correctly 1`] = `
<header
  className="header"
>
  <h1>
    <span>
      Brain
    </span>
    <i
      className="fas fa-brain"
    />
    <span>
      Buster
    </span>
  </h1>
  <div
    className="user-controls"
  >
    <button
      className="toggle-instructions-btn"
      onClick={[Function]}
    >
      Show Instructions
    </button>
    <select>
      <option
        value="default"
      >
        Choose Questions
      </option>
      <option
        value="show-all"
      >
        Show All Questions
      </option>
      <option
        value="show-studylist"
      >
        Study List Only
      </option>
    </select>
  </div>
</header>
`;
